packages/server-wallet/src/wallet/index.ts

```diff
@@ -126,53 +126,22 @@ export class Wallet implements WalletInterface {
   }
 
   async pushMessage(message: AddressedMessage): Result {
-    const channelIds: Bytes32[] = [];
-
-    try {
-      await Channel.transaction(async tx => {
-        for (const ss of message.signedStates || []) {
-          // We ignore unsigned states
-          if (!ss.signatures?.length) return;
-
-          const channelId = calculateChannelId(ss);
-          let channel = await Channel.query(tx)
-            .where('channelId', channelId)
-            .first();
-
-          if (!channel) {
-            const addresses = ss.participants.map(p => p.signingAddress);
-            const signingWallet = await SigningWallet.query(tx)
-              .whereIn('address', addresses)
-              .first();
-
-            if (!signingWallet) {
-              logger.error(
-                {
-                  knownWallets: await SigningWallet.query(tx).select(),
-                  addresses,
-                },
-                'Not in channel'
-              );
-              throw Error('Not in channel');
-            }
-
-            const {address: signingAddress} = signingWallet;
-            const cols: RequiredColumns = {...ss, vars: [addHash(ss)], signingAddress};
-
-            channel = Channel.fromJson(cols);
-            const {channelId} = await Channel.query(tx).insert(channel);
-            channelIds.push(channelId);
-          } else {
-            ss.signatures?.map(sig => channel.addState(ss, sig));
-            await Channel.query(tx).update(channel);
-            channelIds.push(channel.channelId);
-          }
-        }
-      });
-    } catch (err) {
-      logger.error({err}, 'Could not push message');
-      throw err;
-    }
+    await Store.transaction(async () => {
```

Comment:
Do all objectives and all signed states need to be a part of the same transaction?